src/toolbox.ts

```diff
@@ -31,77 +31,84 @@ enum ValidResizeValues {
+};
 
 /**
  * Manager for toolbox. Contains ToolboxTab items.
  */
 export class Toolbox {
-    
     constructor(
         public tabs: ToolboxTab[] = [],
-        public items: ToolboxItem[] = []
-    ) {
-    }
+        public items: ToolboxItem[] = [],
+    ) {}
 
-    public static create_toolbox(ulabel: ULabel, toolbox_item_order: unknown[]) {
+    public static create_toolbox(
+        ulabel: ULabel,
+        toolbox_item_order: unknown[],
+    ) {
         // Grab the default toolbox if one wasn't provided
         if (toolbox_item_order == null) {
-            toolbox_item_order = ulabel.config.toolbox_order
+            toolbox_item_order = ulabel.config.toolbox_order;
         }
 
         // There's no point to having an empty toolbox, so throw an error if the toolbox is empty.
         // The toolbox won't actually break if there aren't any items in the toolbox, so this
         // error isn't strictly neccesary.
         if (toolbox_item_order.length === 0) {
-            throw new Error("No Toolbox Items Given")
+            throw new Error("No Toolbox Items Given");
         }
 
-        this.add_styles()
+        this.add_styles();
 
         const toolbox_instance_list = [];
         // Go through the items in toolbox_item_order and add their instance to the toolbox instance list
         for (let i = 0; i < toolbox_item_order.length; i++) {
-
             let args: object, toolbox_key: number;
 
-            // If the value of toolbox_item_order[i] is a number then that means the it is one of the 
+            // If the value of toolbox_item_order[i] is a number then that means the it is one of the
             // enumerated toolbox items, so set it to the key, otherwise the element must be an array
             // of which the first element of that array must be the enumerated value, and the arguments
             // must be the second value
-            if (typeof(toolbox_item_order[i]) === "number") {
-                toolbox_key = <number> toolbox_item_order[i]
+            if (typeof toolbox_item_order[i] === "number") {
+                toolbox_key = <number>toolbox_item_order[i];
             } else {
                 toolbox_key = toolbox_item_order[i][0];
-                args = toolbox_item_order[i][1]  
+                args = toolbox_item_order[i][1];
             }
 
-            const toolbox_item_class = ulabel.config.toolbox_map.get(toolbox_key);
+            const toolbox_item_class =
+                ulabel.config.toolbox_map.get(toolbox_key);
 
             if (args == null) {
-                toolbox_instance_list.push(new toolbox_item_class(ulabel))
+                toolbox_instance_list.push(new toolbox_item_class(ulabel));
             } else {
-                toolbox_instance_list.push(new toolbox_item_class(ulabel, args))
-            }  
-        }                    
+                toolbox_instance_list.push(
+                    new toolbox_item_class(ulabel, args),
+                );
```

Comment:
this adding a comma at the end is really weird imo

Reply:
For `python` projects I tend to use [black](https://github.com/psf/black), and the easiest way to make it not unwrap (in case I've unwrapped something for readability) is to add a trailing comma. It's so prevalent that I added a different hook ([add-trailing-comma](https://github.com/asottile/add-trailing-comma)) to pre-emptively add it wherever possible so my code doesn't get squished into a 79-character long pile of garbage. 

I've become very used to it as a result. [Here](https://eslint.org/docs/latest/rules/comma-dangle) is eslint's logic for it. In this case it isn't going to change wrapping logic (afaik), so I can try and find a ruleset that doesn't do this if you find it annoying. 

I know it definitely won't add this to JSON as that makes it invalid.

Reply:
i mean its not a deal breaker, just weird to me

Reply:
`prettier` is the one that adds this (see their [docs](https://prettier.io/docs/en/options#trailing-commas)), as the ESLint default for `comma-dangle` is `"never"`. The `prettier` result seems to match the `"always-multiline"` ESLint setting as it doesn't add the trailing comma if it's on the same line (which I think is the best way to go about it). 

`prettier` will still unwrap lines that it can make shorter. They have an option to stop this for markdown (for compatibility reasons), but no options for what we're doing. Not necessarily a deal breaker for me, but worth thinking about.